prody/utilities/catchall.py

```diff
@@ -78,10 +87,18 @@ def calcModeClusters(similarities, n_clusters=0, linkage='all', method='tsne', c
         nclusts = range(2,10,1)
 
     if linkage != 'all':
-        if not isinstance(method, str):
-            raise TypeError('linkage must be an instance of str')
-        if method not in ['ward', 'average', 'complete', 'single']:
-            raise ValueError('linkage must be either \'ward\', \'average\', \'complete\', or \'single\'')
+        if isinstance(linkage, list):
+            if len(linkage) > 4:
```

Comment:
I'd probably do this after the values checking. That way you're checking if something is repeated. It's not relevant how long the list is if there's a wrong value in it.